### PR TITLE
Add JobReadyPods feature gate

### PR DESCRIPTION
**Pull request by alculquicondor**
Add state and description for JobReadyPods feature gate.

The rest of the documentation is on the API reference.

Ref kubernetes/enhancements#2879

/sig apps

### CODE REVIEW

**Comment by chrisnegus** [comment]
@alculquicondor,  I don't see a hold here so is this PR ready to review or are more docs needed?

**Comment by alculquicondor** [comment]
This is ready for review.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -807,6 +808,8 @@ Each feature gate is designed for enabling/disabling a specific feature:
   completions without relying on Pods remaining in the cluster indefinitely.
   The Job controller uses Pod finalizers and a field in the Job status to keep
   track of the finished Pods to count towards completion.
+- `JobReadyPods`: Enables tracking the number of Pods with a Ready condition in
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:811]: nit: I'd link to https://k8s.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

```suggestion
- `JobReadyPods`: Enables tracking the number of Pods that have a `Ready`
  [condition](/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions).
  The count of `Ready` Pods is recorded in
```
```

**Review by sftim — COMMENTED** [review]
Two nits


(As we only expect alpha-quality docs at this point, the PR is fine to merge as-is, providing there's no objection from SIG Apps).

**Comment by alculquicondor** [comment]
Applied the suggestions. Thanks!

/assign @soltysh

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -131,6 +131,7 @@ different Kubernetes components.
 | `JobMutableNodeSchedulingDirectives` | `true` | Beta | 1.23 | |
 | `JobTrackingWithFinalizers` | `false` | Alpha | 1.22 | 1.22 |
 | `JobTrackingWithFinalizers` | `true` | Beta | 1.23 | |
+| `JobReadyPods` | `false` | Alpha | 1.23 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:134]: Move this before line 132

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -807,6 +808,11 @@ Each feature gate is designed for enabling/disabling a specific feature:
   completions without relying on Pods remaining in the cluster indefinitely.
   The Job controller uses Pod finalizers and a field in the Job status to keep
   track of the finished Pods to count towards completion.
+- `JobReadyPods`: Enables tracking the number of Pods that have a `Ready`
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:811]: move this before line 807

**Comment by alculquicondor** [comment]
Fixed order

**Comment by shannonxtreme** [comment]
@alculquicondor did the order fixing that Qiming suggested work? I don't see a change (not sure if on my end)

**Comment by alculquicondor** [comment]
Sorry, I forgot to push :|

**Comment by shannonxtreme** [comment]
Thank you :) This looks great, thanks for the PR!

/lgtm

**Comment by chrisnegus** [comment]
@alculquicondor Is this ready to merge from your perspective?

**Comment by alculquicondor** [comment]
yes

**Comment by chrisnegus** [comment]
@alculquicondor Thanks!

**Review by soltysh — APPROVED** [review]
/lgtm
from sig-apps pov

**Comment by chrisnegus** [comment]
> /lgtm from sig-apps pov

Thanks @soltysh !

**Pull request merged**